src/targets.rs

```diff
@@ -96,6 +96,63 @@ impl Default for InitializationConfig {
     }
 }
 
+#[derive(Eq)]
+pub struct TargetTriple {
+    pub(crate) target_triple: LLVMStringOrRaw,
+}
+
+impl TargetTriple {
+    pub(crate) fn new_owned(target_triple: *const ::libc::c_char) -> TargetTriple {
+        debug_assert!(!target_triple.is_null());
+
+        TargetTriple {
+            target_triple: LLVMStringOrRaw::Owned(LLVMString::new(target_triple)),
+        }
+    }
+
+    pub(crate) fn new_borrowed(target_triple: *const ::libc::c_char) -> TargetTriple {
+        debug_assert!(!target_triple.is_null());
+
+        TargetTriple {
+            target_triple: LLVMStringOrRaw::Borrowed(target_triple),
+        }
+    }
+
+    pub fn create(target_triple: &str) -> TargetTriple {
```

Comment:
It's not really clear to me what advantage `TargetTriple` has over just passing a string since you can just give `create` any old invalid string? It seems like the same problem as before?

Reply:
I was thinking of this as an application of [newtype pattern](https://doc.rust-lang.org/1.0.0/style/features/types/newtype.html). The advantage is just that it can help people avoid mixing up strings in their code that represent different things.

One could theoretically add some validation in the future though I'm not actually sure if it's feasible.

Reply:
I'm not sure that it is feasible to add validation because you can create custom triples I think? Or at least Rust lets you define triple specifications. Not sure how that works when it gets to LLVM

Reply:
I need to look through this again and think on it some more

Reply:
I think there's value in a newtype when you can guarantee that only valid `TargetTriple` are generated, which we cannot

Reply:
I see both perspectives here, so I'm not super inclined to argue hard for the newtype. It's my personal inclination, but you've got a point. Just let me know what you decide on this so I can finish up the PR and we can merge it.